### ISSUE TITLE
Update setup.py to confirm python requirements

### ISSUE THREAD

**Issue by jeromekelleher** [tskit-dev/tskit#520]
Setup.py currently states that we need python >= 3.4, but we've recently started using f-strings. 

Since Debian stable is currently on 3.7 and f-strings are pretty useful, I think we're OK to push our requirements up to 3.6. Happy to be convinced otherwise, although actually making sure that older versions of Python are supported is not trivial and pushes our CI requirements up a lot.

**Comment by brianzhang01** [comment]
I was on Python 3.5 when I encountered and reported this, but I guess it's time I upgraded. So the change works for me, but we should document it and maybe be extra explicit about what's going on.

The errors I got were:
- From flake8: `E999 SyntaxError: invalid syntax` at all the places with f-strings
- From nose tests: 

```
Failure: SyntaxError (invalid syntax (formats.py, line 247)) ... ERROR
Failure: SyntaxError (invalid syntax (formats.py, line 247)) ... ERROR
...
    raise ValueError(f"Version {format_version} not supported for loading")
```

I didn't do the pip install of development tskit, but some issues might arise there as well. Would be good to catch these all early and notify the user.
